### PR TITLE
Add extra quotes to fix paths with spaces in them

### PR DESCRIPTION
**Pull request by jonathanfrawley**
- Fixes scala/bug#10756

### CODE REVIEW

File: src/compiler/scala/tools/ant/templates/tool-unix.tmpl [hunk 1]
```diff
@@ -13,7 +13,7 @@ findScalaHome () {
   local source="${BASH_SOURCE[0]}"
   while [ -h "$source" ] ; do
     local linked="$(readlink "$source")"
-    local dir="$( cd -P $(dirname "$source") && cd -P $(dirname "$linked") && pwd )"
+    local dir="$( "cd -P $(dirname "$source")" && cd -P "$(dirname "$linked")" && pwd )"
```
Comment (som-snytt) [inline src/compiler/scala/tools/ant/templates/tool-unix.tmpl]: The first quote is misplaced, and is intended before the next dollar?

**Comment by som-snytt** [comment]
I reproduced by untar to dir with a space, then linking to `"my scala"/bin/scala`.

The extra quotes work for me on MacOS but not Ubuntu:
```
amarki@amarki-462836:~/tmp$ ./sc
./sc: line 16: cd: my: No such file or directory
Error: Could not find or load main class scala.tools.nsc.MainGenericRunner
amarki@amarki-462836:~/tmp$ vi "my scala"/bin/scala
amarki@amarki-462836:~/tmp$ ./sc
./sc: line 20: cd: ./my scala/bin
/home/amarki/tmp/my scala/bin/..: No such file or directory
Error: Could not find or load main class scala.tools.nsc.MainGenericRunner
amarki@amarki-462836:~/tmp$ ls -l sc
lrwxrwxrwx 1 amarki amarki 18 Mar  6 09:10 sc -> my scala/bin/scala

```

**Comment by jonathanfrawley** [comment]
Apologies, I transcribed it from an external script that I was testing with and I put the quote in the wrong place in the result.

I have corrected the mistake and I will test it as you mentioned on MacOS and Linux when it has finished building.

**Comment by jonathanfrawley** [comment]
Added a comment on the issue with a script to test this. Verified to work on Fedora Linux and MacOS. https://github.com/scala/bug/issues/10756

**Comment by som-snytt** [comment]
I wonder if the PR could go against 2.12.x and get into 2.12.5 this week.

The change is probably benign at worst, but I wish I knew why it doesn't work for me on Ubuntu. I'll also try a Red Hat server at work, just for kicks.

Someone will be in my shoes and will say it still doesn't work, or works worse.

**Comment by lrytz** [comment]
Tried on a debian 8 and the fix works for me. 2.12.5 is fine with me if it's reviewed and ready soon - would be interested to hear what you find on your ubuntu, @som-snytt

**Review by som-snytt — APPROVED** [review]
Thanks! Can we retarget to 2.12.x by editing the PR?

**Comment by SethTisue** [comment]
I've changed the milestone; @df3n5 can you change the PR target branch and rebase onto current 2.12.x? (if this isn't something you've done before and have Git/GitHub questions about it, fire away.)

**Comment by jonathanfrawley** [comment]
@SethTisue I have done this now. Github will not let me change the source branch name which is a bit unfortunate but I suppose this is just cosmetic. If you would like me to open up a new pull request to fix this let me know.
@som-snytt Did you check my revised version? I have just checked this on Ubuntu 16.04, Fedora 27 and MacOS and it works fine for me. My first commit did not work however so that might have been what you were testing with. I used this to test for reference:

```bash
#!/bin/bash
set -e
mkdir -p build
rm -vrf build/pack
sbt dist/mkPack

pushd build/pack
cp -rvf bin/ bad\ bin
mkdir -v bad\ linked\ bin
ln -sv `pwd`/bad\ bin/scala `pwd`/bad\ linked\ bin/scala\ link
bad\ linked\ bin/scala\ link -e 'val e = 123; println(s"e is : ${e}")'
popd
```

**Comment by lrytz** [comment]
@df3n5 thanks for rebasing - to speed things up (for me) I squashed the commits and re-submitted in https://github.com/scala/scala/pull/6399. You're still the commit author :)

**Comment by jonathanfrawley** [comment]
@lrytz Cool, thanks! I should have asked if you wanted it squashed as well. I'll do that for my next contribution.

**Pull request closed without merging**